src/main/java/com/labsynch/labseer/api/ApiSaltController.java

```diff
@@ -82,6 +95,191 @@ public ResponseEntity<String> showJson(@PathVariable("id") Long id) {
 		return new ResponseEntity<String>(salt.toJson(), headers, HttpStatus.OK);
 	}
 
+	@RequestMapping(value = "/edit/{id}", method = RequestMethod.PUT, headers = "Accept=application/json")
+	@ResponseBody
+	public ResponseEntity<String> editSalt(@PathVariable("id") Long id, @RequestBody String json) {
+		Salt oldSalt = Salt.findSalt(id);
+		HttpHeaders headers = new HttpHeaders();
+		headers.add("Content-Type", "application/text; charset=utf-8");
+		headers.add("Access-Control-Allow-Origin", "*");
+		headers.add("Access-Control-Allow-Headers", "Content-Type");
+		headers.add("Cache-Control", "no-store, no-cache, must-revalidate"); // HTTP 1.1
+		headers.add("Pragma", "no-cache"); // HTTP 1.0
+		headers.setExpires(0); // Expire the cache
+		if (oldSalt == null) {
+			return new ResponseEntity<String>("Salt could not be found", headers, HttpStatus.NOT_FOUND);
+		}
+		else
+		{
+			try {
+				ObjectMapper objectMapper = new ObjectMapper();
+
+				JsonNode rootNode = objectMapper.readTree(json);
+				JsonNode dryRunNode = rootNode.path("dryrun");
+				boolean dryrun = dryRunNode.asBoolean();
+
+				rootNode = objectMapper.readTree(json);
+				JsonNode saltNode = rootNode.path("saltJSON");
+				Salt newSalt = Salt.fromJsonToSalt(saltNode.toString());
+
+				ArrayList<ErrorMessage> warnings = new ArrayList<ErrorMessage>();
+
+				if(!newSalt.getAbbrev().equals(oldSalt.getAbbrev())){
+					ErrorMessage warning = new ErrorMessage();
+					warning.setLevel("warning");
+					warning.setMessage("The abbreviation of this salt will be changed.");
+					warnings.add(warning);
+				}
+				if(!newSalt.getName().equals(oldSalt.getName())){
+					ErrorMessage warning = new ErrorMessage();
+					warning.setLevel("warning");
+					warning.setMessage("The name of this salt will be changed.");
+					warnings.add(warning);
+				}
+
+				newSalt.setAbbrev(newSalt.getAbbrev().trim());
+				newSalt.setName(newSalt.getName().trim());
+				newSalt.setFormula(saltStructureService.calculateFormula(newSalt));
+
+				// This doesn't always work; Might need to convert to mol and then compare
```

Comment:
I will want to implement a CompareTo method of some sort for salt structs

Reply:
Can we just calculate weight & charge, and skip comparing structures? Structure comparison is a rabbit hole, and we should avoid it if we can.

Reply:
So, just to confirm, there should be no alert if someone makes an isomer of a salt?

Reply:
Confirmed. No alert if someone makes an isomer of a salt. We can always add something like that later, but for now let's keep it simple.